werkzeug/datastructures.py

```diff
@@ -1153,9 +1153,15 @@ def add(self, _key, _value, **kw):
         The keyword argument dumping uses :func:`dump_options_header`
         behind the scenes.
 
+        In some cases, it would be needed to ensure the order of parameters.
+        For those cases, a list of ``(key, value)`` tuples can be passed in
+        *args.
+
         .. versionadded:: 0.4.1
             keyword arguments were added for :mod:`wsgiref` compatibility.
         """
+        if args:
+            _value = dump_options_header(_value, args)
```

Comment:
I agree with the rest of changes but I won't have time today to work on this. I'll try to fix everything tomorrow or Monday the latest